contracts/staking/contracts/src/staking_pools/MixinStakingPoolMakers.sol

```diff
@@ -0,0 +1,210 @@
+    /// Note also that the maker must have previously called joinStakingPoolAsMaker.
+    /// @param poolId Unique id of pool.
+    /// @param makerAddress Address of maker.
+    function addMakerToStakingPool(
+        bytes32 poolId,
+        address makerAddress
+    )
+        external
+        onlyStakingPoolOperator(poolId)
+    {
+        _addMakerToStakingPool(poolId, makerAddress);
+    }
+
+    /// @dev Removes a maker from a staking pool. Note that this is only callable by the pool operator or maker.
+    /// Note also that the maker does not have to *agree* to leave the pool; this action is
+    /// at the sole discretion of the pool operator.
+    /// @param poolId Unique id of pool.
+    /// @param makerAddress Address of maker.
+    function removeMakerFromStakingPool(
+        bytes32 poolId,
+        address makerAddress
+    )
+        external
+        onlyStakingPoolOperatorOrMaker(poolId, makerAddress)
+    {
+        bytes32 makerPoolId = getStakingPoolIdOfMaker(makerAddress);
+        if (makerPoolId != poolId) {
+            LibRichErrors.rrevert(LibStakingRichErrors.MakerPoolAssignmentError(
+                LibStakingRichErrors.MakerPoolAssignmentErrorCodes.MakerAddressNotRegistered,
+                makerAddress,
+                makerPoolId
+            ));
+        }
+
+        // remove the pool and confirmation from the maker status
+        IStructs.MakerPoolJoinStatus memory poolJoinStatus = IStructs.MakerPoolJoinStatus({
+            poolId: NIL_POOL_ID,
+            confirmed: false
+        });
+        poolJoinedByMakerAddress[makerAddress] = poolJoinStatus;
+        poolById[poolId].numberOfMakers = uint256(poolById[poolId].numberOfMakers).safeSub(1).downcastToUint32();
+
+        // Maker has been removed from the pool`
+        emit MakerRemovedFromStakingPool(
+            poolId,
+            makerAddress
+        );
+    }
+
+    /// @dev Returns the pool id of the input maker.
+    /// @param makerAddress Address of maker
+    /// @return Pool id, nil if maker is not yet assigned to a pool.
+    function getStakingPoolIdOfMaker(address makerAddress)
+        public
+        view
+        returns (bytes32)
+    {
+        if (isMakerAssignedToStakingPool(makerAddress)) {
+            return poolJoinedByMakerAddress[makerAddress].poolId;
+        } else {
+            return NIL_POOL_ID;
+        }
+    }
+
+    /// @dev Returns true iff the maker is assigned to a staking pool.
+    /// @param makerAddress Address of maker
+    /// @return True iff assigned.
+    function isMakerAssignedToStakingPool(address makerAddress)
+        public
+        view
+        returns (bool)
+    {
+        return poolJoinedByMakerAddress[makerAddress].confirmed;
+    }
+
+    /// @dev Returns the current number of makers in a given pool.
+    /// @param poolId Unique id of pool.
+    /// @return Size of pool.
+    function getNumberOfMakersInStakingPool(bytes32 poolId)
```

Comment:
I'm kind of meh on having all of these getters. The rest of them are actually pretty frequently used so I'm fine leaving them in, but we should remove this one. When we're using getters for individual properties, it is easy to incur extra costs from sloads (example below).